buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -1006,6 +1006,13 @@ private AbstractByteBuf rootParent() {
             throw new IllegalReferenceCountException();
         }
 
+        private ByteBuffer getTmpNioBuf() {
+            if (tmpNioBuf == null) {
+                tmpNioBuf = rootParent().internalNioBuffer(adjustment, length).slice();
```

Comment:
No `cas` here?

Reply:
It is supposed to have single threaded usage afaik

Reply:
I might not but it should not matter imho

Reply:
If it matters - users relies on an existing previous impl detail we don't enforce eheh

Reply:
this code will be idempotent as long as root parent or capacity hasn't changed… so, idempotent in the face of concurrent read access